src/ewatercycle/_forcings/caravan.py

```diff
@@ -0,0 +1,240 @@
+                 basin_id: str,
+                 directory: str | Path,
+                 variables: tuple[str, ...] = (),
+                 shape: str | Path | None = None,
+                 **kwargs) -> "Caravan":
+        """Retrieve caravan for a model.
+
+        Args:
+            start_time: Start time of forcing in UTC and ISO format string e.g.
+                'YYYY-MM-DDTHH:MM:SSZ'.
+            end_time: nd time of forcing in UTC and ISO format string e.g.
+                'YYYY-MM-DDTHH:MM:SSZ'.
+            basin_id: Caravan basin_id
+            directory: Directory in which forcing should be written.
+                If not given will create timestamped directory.
+            variables: Variables which are needed for model,
+                if not specified will default to all.
+            shape: (Optional) Path to a shape file.
+                If none is specified, will be downloaded automatically.
+            **kwargs
+        """
+        dataset = basin_id.split('_')[0]
+        ds = xr.open_dataset(
+            f"{OPENDAP_URL}{dataset}.nc")
+        ds_basin = ds.sel(basin_id=basin_id.encode())
+        ds_basin_time = crop_ds(ds_basin, start_time, end_time)
+
+        if shape is None:
+            shape = get_shapefiles(Path(directory), basin_id)
+
+        if 'lat' not in variables and 'lon' not in variables:
+            shape_obj = shapereader.Reader(shape)
+            centre = next(shape_obj.geometries()).centroid
+            ds_basin_time.coords.update({'lat': centre.y,
+                                         'lon': centre.x
+                                         })
+        if variables == ():
+            variables = (ds_basin_time.data_vars.keys())
+
+        # only return the properties which are also in property vars
+        properties = tuple(set(variables).intersection(set(PROPERTY_VARS)))
+        variables = tuple(
+            set(variables).difference(set(PROPERTY_VARS)).intersection(
+              set(RENAME_ERA5.keys())
+            )) # only take the vars also in Rename dict
+
+        for prop in properties:
+            ds_basin_time.coords.update({prop: ds_basin_time[prop].to_numpy()})
+
+        ds_basin_time = ds_basin_time.rename(RENAME_ERA5)
+        variables = ([RENAME_ERA5[var] for var in variables])
+
+        for temp in ['tas','tasmin','tasmax']:
+            ds_basin_time[temp].attrs.update({'height':'2m'})
+
+        for var in variables:
+            ds_basin_time[var].to_netcdf(Path(directory) / f'{basin_id}_{start_time}_{end_time}_{var}.nc')
+
+        forcing = cls(
+            directory=Path(directory),
+            start_time=start_time,
+            end_time=end_time,
+            shape=Path(shape),
+            filenames={
+                var: f'{basin_id}_{start_time}_{end_time}_{var}.nc' for var in variables
+            }
+        )
+        forcing.save()
+        return forcing
+
+
+class LumpedCaravanForcing(Caravan, LumpedUserForcing):  # type: ignore[misc]
```

Comment:
You can move this to the top. There won't be a distributed caravan forcing, so you can just inherit in the class containing the implementations.

Also, as there won't be a DistributedCaravanForcing, it might be better to just name it `CaravanForcing`.